### ISSUE TITLE
Problem with "embed with"

### ISSUE THREAD

**Issue by rikschennink** [twigjs/twig.js#386]
It's pretty straightforward:

This works:

``` twig
{% embed '_partials/validator.twig' with { 'id':field.id, 'errors':field.errors } %}
```

This does not:

``` twig
{% embed '_partials/validator.twig' with 
{
'id':field.id,
'errors':field.errors
}
%}
```

But with the include statement it does:

``` twig
{% include '_partials/form-item.twig' with
{
'id':field.id,
'label':field.label
}
%}
```

**Comment by derekreesemediacurrent** [comment]
Confirming. In addition, substituting out blocks with empty/null variables causes an error:
```
{% embed '@components/media-item/components.media-item.twig' with { 'classes': 'media-item', 'title': item.title, 'url': item.url, 'image': item.image, 'label': item.label, 'type': item.type  } only %}
          {% block text %}
            <div class="media-item__field"><a href="#">{{ fieldText }}</a></div>
          {% endblock text %}
        {% endembed %}
```
I think from what I gather in the Twig docs that the lack of the fieldText variable should not stop the block from displaying the markup in this situation.

**Comment by olets** [comment]
@derekreesemediacurrent I'm not following your "in addition." What here is empty or null, and what's breaking? (Maybe a more minimal example?)

Tangent: ooh did you create an @components namespace or something?? And you have it working in twig.js? If that's correct, I'd love to hear how you did that. Don't want to swamp this thread… if it's lengthy, maybe dm me on twig.js gitter

**Comment by olets** [comment]
Looks to me like this is more specifically any time there's a line break between the `{` and the `}` of an `embed` tag's `with` (i.e. twigjs errors even if there's only one line break in there, and it can be anywhere in there)

**Comment by derekreesemediacurrent** [comment]
@olets The null variable is `fieldText`, which isn't passed in the `with` statement part of the embed. Shouldn't Twig.js should treat it just like any other variable that isn't provided a value?
```
{% embed '@components/media-item/components.media-item.twig' with { 'classes': 'media-item' } only %}
  {% block text %}
    <div class="media-item__field"><a href="#">{{ fieldText }}</a></div>
  {% endblock text %}
{% endembed %}
```

For namespaces, see https://github.com/twigjs/twig.js/pull/195

**Comment by olets** [comment]
Ah okay didn't know if `fieldText` had been set somewhere else. You're definitely right about how it should work - e.g. http://twigfiddle.com/zz6agb. Can you check that in twig.js? It's working for me

Continuing the namespaces tangent ;) Sweet! Where are you adding that variable?

**Comment by jpdevries** [comment]
I've run into this issue as well
http://twigfiddle.com/j02f4n

```twig
{% embed "label.twig" with {
    'for': 'cat'
  } only %}
    {% block innerHTML %}
      A cool cat
    {% endblock %}
{% endembed %}
```

```twig
<label{% if accesskey %} accesskey="{{accesskey}}"{% endif %}{% if class %} class="{{class}}"{% endif %}{% if contenteditable %} contenteditable="{{contenteditable}}"{% endif %}{% if contextmenu %} contextmenu="{{contextmenu}}"{% endif %}{% if dir %} dir="{{dir}}"{% endif %}{% if draggable %} draggable="{{draggable}}"{% endif %}{% if dropzone %} dropzone="{{dropzone}}"{% endif %}{% if for %} for="{{for}}"{% endif %}{% if form %} form="{{form}}"{% endif %}{% if hidden %} hidden="{{hidden}}"{% endif %}{% if id %} id="{{id}}"{% endif %}{% if lang %} lang="{{lang}}"{% endif %}{% if spellcheck %} spellcheck="{{spellcheck}}"{% endif %}{% if style %} style="{{style}}"{% endif %}{% if tabindex %} tabindex="{{tabindex}}"{% endif %}{% if title %} title="{{title}}"{% endif %}{% if translate %} translate="{{translate}}"{% endif %}>
  {% block innerHTML %}{% endblock %}
</label>
```

<details>
  <summary>Console Error</summary>
  ```bash
Error compiling twig template /Users/jP/Sites/twag/views/index.twig: 
TwigException: Unable to parse 'embed "../twag/label.twig" with {
    'for': 'cat'
  } only'
Error parsing twig template /Users/jP/Sites/twag/views/index.twig: 
TypeError: Cannot read property 'forEach' of undefined
    at Object.Twig.forEach (/Users/jP/Sites/twag/node_modules/twig/twig.js:151:24)
    at Twig.parse (/Users/jP/Sites/twag/node_modules/twig/twig.js:830:19)
    at Twig.Template.render (/Users/jP/Sites/twag/node_modules/twig/twig.js:1267:30)
    at params.load (/Users/jP/Sites/twag/node_modules/twig/twig.js:7009:41)
    at loadTemplateFn (/Users/jP/Sites/twag/node_modules/twig/twig.js:5374:18)
    at tryToString (fs.js:455:3)
    at FSReqWrap.readFileAfterClose [as oncomplete] (fs.js:442:12)
  ```
</details>

**Comment by johny-gog** [comment]
It seems fixed in v0.10.0
